actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -467,15 +469,29 @@ def initialize(parent_jar)
 
       def [](name)
         if data = @parent_jar[name.to_s]
-          parse name, data
+          value = parse name, data
+          if Cookies.cookies_have_purpose
+            if value["purpose"].eql?(name.to_s)
+              value["value"]
+            else
+              nil
+            end
+          else 
+            value
+          end
```

Comment:
it's nothing about nested conditionals there. 

```
if data = @parent_jar[name.to_s]
  value = parse name, data
  if Cookies.cookies_have_purpose
    if value["purpose"].eql?(name.to_s)
      value["value"]
    else
      nil
    end
  else 
    value
  end
end
```

Because 3 levels here makes code a little bit more difficult to read.